### ISSUE TITLE
@each-loop is malforming multidimensional sass-lists with only one nested list

### ISSUE THREAD

**Issue by vsxed** [sass/sass#1702]
Hello there,
today i managed to reproduce a bug / a strange behavior with sass lists in combination with an @each-loop.
# **tl;dr**:

![cczm7rbw4aa8djg](https://cloud.githubusercontent.com/assets/1070998/7213056/23f4d6e0-e573-11e4-9430-5782f4cb7ee9.png)
## Example SCSS:

![bildschirmfoto 2015-04-18 um 01 42 00](https://cloud.githubusercontent.com/assets/1070998/7212740/3e6d1386-e56c-11e4-8022-1148a63ac393.png)
## Output: CSS

![bildschirmfoto 2015-04-18 um 01 42 12](https://cloud.githubusercontent.com/assets/1070998/7212741/409ff902-e56c-11e4-9d32-2ce0061259b3.png)

As you can see, there's a problem when you got a (so thought and structured) multidimensional sass list with only one nested list inside: 
- if there's a seperating comma after the nested list, the output code will be given as everybody would expect it. 
- if this seperator is missing, the @each-loop seems to be malforming our nested list into a single list with single comma-seperated values -> the output code is wrong (see compiled preview above)

When using a real multidimensional nested sass-list with more than 1 list, everything is all right. in both cases: with and without a seperating comma at the end of the last nested list.

As you can see, i registered this behavior while developing a font face loop using sass lists, where also a mixin is used within the loop. (i cropped the original function, for better readability :+1: )
## Try it out yourself

here's a link to the example scss on sassmeister:
http://sassmeister.com/gist/72f65ceac6d0f7847045

It's possible to reproduce this with:
- (CLI) Sass 3.4.10+ (Selective Steve)
- (CLI) Sass 3.4.13 (Selective Steve) 

Operating System: 
- Mac OS X 10.10
- Mac OS X 10.9.5

**Comment by nex3** [comment]
This is intended. It's important to understand that `(`/`)` are **not** list delimiters in Sass; they signal precedence, just like they do in other languages. In order to be compatible with CSS syntax, lists are defined only by their delimiters, whether it's a space or a comma. This means that the outer parenthesis in `$list1` in your example don't do anything.

When we introduced lists, one of the unfortunate consequences of this syntax was that there was no way to write a single-element list. In 3.3.0, though, we added a way to do so: if you add a comma after a single element, Sass now considers that to be a single-element list. So `(foo,)` is now a list containing the element `foo`, which is how `$list2` works.

**Comment by vsxed** [comment]
Nice, thanks! Didn't know that!
